### DISCUSSION TITLE
What new features are the most needed ?

### DISCUSSION THREAD

**Discussion by emanuelef** [emanuelef/daily-stars-explorer#80] [Polls]
I'm looking to gather insights on potential new features for this project.
While I'm always delighted with contributions through PRs, I've outlined some options in this poll. If you have additional suggestions, feel free to let me know.
Appreciate your input!
Thanks.

**Comment by Kerollmops** [comment]
I would love to vote for more than a single feature 😞

**Reply by emanuelef** [reply]
I cannot find the option in the poll to select more.
In the end this is just a way to understand what can be added next, maybe there's something that is not in the options presented.
Happy to hear what could be helpful to add.

**Comment by Kerollmops** [comment]
One fun thing I am interested in is graph annotations: add a text, most of the time an URL, to the different spikes we can see on the graph.

The remaining questions are:
- Who can do that?
- If anyone can, then it should probably be local and not visible to others.
- A better approach could be to only let the owners of the repo add those comments. Maybe API keys can give the account information and organizations?

**Reply by emanuelef** [reply]
Interesting idea!
I was thinking to correlated the spikes to articles mentioning that repo on Reddit, Medium or YouTube. But not sure is feasible in terms of the APIs exposed by those services. Having that we could see if a spike was related to a new article or video on YouTube and it will be automated.

**Reply by Kerollmops** [reply]
That's kind of what I thought about but as you explained, it could be complex to implement. A much easier feature could be to let users do that "by hand". What do you think?

**Reply by emanuelef** [reply]
That would be a nice feature, the options you are suggesting will require to persist that data and some form of authentication.
At the moment I'm only updating the dependencies but not actively working on the project, very happy if someone wants to start contributing.

**Reply by emanuelef** [reply]
I had some time to try integrating HN, it's something very rough at the moment.
I get the articles from HN API (I didn't spent too much time in tuning al the query params, docs here: https://hn.algolia.com/api)
An example choosing a random project on GH ;-) :  https://emanuelef.dynv6.net:8090/hackernews?query=meilisearch

As this is very experimental I changed the Derivative filter temporarily to show the first 20.
Clicking on the time marker opens the HN article.

![Screenshot 2024-08-04 at 16 36 43](https://github.com/user-attachments/assets/a6d9eded-1399-4b04-954f-c38d44afba41)

Do they make sense ? Suggestions on a better query and post filtering to get only relevant results ?

I plan to try with Reddit and some YouTube channels later, not sure when I get some time next.

In terms of UI for representing those articles on the timeline [FusionTime annotations](https://www.fusioncharts.com/fusiontime/examples/annotating-single-data-point?framework=javascript) would be more appropriate, something to try later.

**Reply by Kerollmops** [reply]
Hey @emanuelef 👋

I'm impressed by the quality of work already, that's actually very good and seems already quite useful 🤩

> Do they make sense ?

Yes, they make absolute sense! We know the first one by heart!

> Suggestions on a better query and post filtering to get only relevant results ?

Probably not, I'm pretty sure the search results are provided by Algolia which only support keyword search. Using the repository name seems ok to me but you only tried with Meilisearch, it will be hard to track Elasticsearch for example...

> I plan to try with Reddit and some YouTube channels later, not sure when I get some time next.

Looking forward to see those APIs plugged in. However, the last spike you see on the Meilisearch graph is for sure thanks to a Twitter post of Arvid Kahl. I know that Twitter is private though...

Ho! And adding a link to the HN story or future videos/blog posts will be perfect!

> In terms of UI for representing those articles on the timeline FusionTime annotations would be more appropriate, something to try later.

I find it great for a first release but using FusionTime annotations will make it even better for sure!

Can you release a first version in public right now with only Hackernews so that I can try a couple repositories already?

You rocks 🎸

**Reply by emanuelef** [reply]
Thanks, that change is already on the public release.
When clicking on the time marker it opens the URL, it's not clear though. I'll probably add a section outside of the graph to show the details when clicking or hovering over the time marker, that will make possible to add a clickable link.

On the search now it's just using the repo name, I wonder if using the GH URL or owner/repo could give better results, especially for cases where the repo name is a common word used in other contexts. 
I'll probably go on with Reddit integration but that will be next week end :-)

**Reply by Kerollmops** [reply]
Hey @emanuelef 👋

I tried more repositories, and there is something strange with [the _strapi/strapi_ repo](https://github.com/strapi/strapi). Look at the first article just before the big spike, it's about HIV attacks, absolutely not related to strapi or web or anything. Do you think the search request matches the _strains_ word? It seems that the first article is also not matching _strapi_ but _Scraping_ 🤔

Ho! And it would also be great to have support for ProductHunt.

**Reply by emanuelef** [reply]
[http://hn.algolia.com/api/v1/search?query=strapi&hitsPerPage=200&numericFilters=points>300](http://hn.algolia.com/api/v1/search?query=strapi&hitsPerPage=200&numericFilters=points%3E300)

one of the entries has in matchedWords strapi, which is surprising.
I wanted to filter out by matchedWords not containing strapi, but I guess I'd need to do that from the title itself...

```json
   {
      "_highlightResult": {
        "author": {
          "matchLevel": "none",
          "matchedWords": [],
          "value": "gh_hammour"
        },
        "title": {
          "fullyHighlighted": false,
          "matchLevel": "full",
          "matchedWords": [
            "strapi"
          ],
          "value": "Saint Michael Sword: Are the cathedrals really on a <em>strai</em>ght line?"
        },
        "url": {
          "matchLevel": "none",
          "matchedWords": [],
          "value": "https://geospatial.netlify.app/posts/gds-2024-04-20-cathedrals/"
        }
      },
      "_tags": [
        "story",
        "author_gh_hammour",
        "story_40595499"
      ],
      "author": "gh_hammour",
      "children": [
        40595595,
        40595806,
        40595820,
        40595895,
        40595900,
        40595931,
        40595976,
        40595979,
        40596002,
        40596008,
        40596098,
        40596131,
        40596144,
        40596199,
        40596227,
        40596280,
        40596291,
        40596295,
        40596297,
        40596434,
        40596465,
        40596495,
        40596629,
        40596717,
        40596791,
        40596837,
        40596906,
        40597039,
        40597107,
        40597151,
        40597206,
        40597314,
        40597495,
        40597720,
        40597731,
        40598294,
        40598689,
        40598959,
        40599272,
        40600578,
        40604597,
        40605592,
        40605778,
        40605859,
        40606502,
        40606911,
        40607534,
        40609054,
        40609509,
        40610791,
        40614614,
        40615103
      ],
      "created_at": "2024-06-06T09:48:59Z",
      "created_at_i": 1717667339,
      "num_comments": 251,
      "objectID": "40595499",
      "points": 372,
      "story_id": 40595499,
      "title": "Saint Michael Sword: Are the cathedrals really on a straight line?",
      "updated_at": "2024-07-22T14:53:22Z",
      "url": "https://geospatial.netlify.app/posts/gds-2024-04-20-cathedrals/"
    }

**Reply by emanuelef** [reply]
I added the three feeds fetch (HN, Reddit and YouTube).
They all need some tuning cause I see many false positives, I will try to read more about the search options for each of them.
So far I tried on some to limit the results to the ones with the repo name present in the title, but there's still more that can be done to make the results relevant. Something I was considering is to add `github` to the search query.
Any help on how to improve search with better query options or post filtering is more than welcome.
